drivers/lora/sx12xx_common.c

```diff
@@ -73,6 +76,15 @@ static void sx12xx_ev_tx_done(void)
 	if (MODEM_FREE(dev_data)) {
 		Radio.Sleep();
 	}
+
+#ifdef CONFIG_LORA_ASYNC
+	if (dev_data.tx_signal) {
```

Comment:
There seems to be a race in this code since you are releasing the modem before reading the signal. You'll need to keep a local copy of the signal and clear it the dev_data copy before releasing the modem.

Reply:
Yes. The release does need to stay after the sleep so that the signal can setup another transmission.